### PR TITLE
(RSP-1760) Stop passing press event to onClear 

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes https://jira.corp.adobe.com/browse/RSP-1760

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Go to SearchField onClear story. Test clearing the field with ESC key and by clicking the 'x' button in the field. Verify that the clear action in the storybook action panel is called with `[ ]` meaning it is passed nothing

### CODE REVIEW

File: packages/@react-spectrum/buttongroup/src/ButtonGroup.tsx [hunk 1]
```diff
@@ -61,6 +61,8 @@ function ButtonGroup(props: SpectrumButtonGroupProps, ref: DOMRef<HTMLDivElement
     if (!dirty) {
       setDirty(true);
     }
+  // Don't add dirty to dep array since it will cause infinite loop
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (snowystinger) [inline packages/@react-spectrum/buttongroup/src/ButtonGroup.tsx:65]: why/how did this end up in here?

File: packages/@react-aria/searchfield/test/useSearchField.test.js [hunk 2]
```diff
@@ -134,6 +134,7 @@ describe('useSearchField hook', () => {
         expect(ref.current.focus).toHaveBeenCalledTimes(1);
         // Verify that props.onClear is triggered as well with the same event
         expect(onClear).toHaveBeenCalledTimes(1);
+        expect(onClear).not.toHaveBeenCalledWith(mockEvent);
```
Comment (snowystinger) [inline packages/@react-aria/searchfield/test/useSearchField.test.js:137]: can you assert that it was called with nothing?
expect(onClear).toHaveBeenCalledWith()

or maybe there's a matcher that means nothing?

File: packages/@react-spectrum/buttongroup/src/ButtonGroup.tsx (hunk 1 above)
Comment (LFDanLu) [reply packages/@react-spectrum/buttongroup/src/ButtonGroup.tsx:65]: Noticed it when doing a lint locally, and remembered a suggestion (forget from who) to disabled the lint warning so calling `--fix` wouldn't break everything by putting `dirty` into the dep array 

Coulda sworn we had added this before, but dunno why it is here again

File: packages/@react-aria/searchfield/test/useSearchField.test.js (hunk 2 above)
Comment (LFDanLu) [reply packages/@react-aria/searchfield/test/useSearchField.test.js:137]: ooo, lemme try that, makes more sense

File: packages/@react-spectrum/buttongroup/src/ButtonGroup.tsx (hunk 1 above)
Comment (snowystinger) [reply packages/@react-spectrum/buttongroup/src/ButtonGroup.tsx:65]: i thought we had too, haha, well, this is fine

**Pull request merged**